src/main/java/frc/utils/pid/PIDUtils.java

```diff
@@ -0,0 +1,100 @@
+package frc.utils.pid;
+
+import com.revrobotics.SparkPIDController;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+/**
+ * Contains helper methods for simplifying and abstracting code related to PID controllers.
+ *
+ * @see frc.utils.pid.PIDConfig
+ * @see frc.utils.pid.PIDParam
+ */
+public final class PIDUtils {
+
+  private PIDUtils() {}
+
+  /**
+   * Maps a {@link frc.utils.pid.PIDParam} to the corresponding setter method of a {@link
+   * com.revrobotics.SparkPIDController}.
+   *
+   * <p>Example: {@code getCorrespondingSetMethod(controller, PIDParam.P)} will return a {@link
+   * java.util.function.Consumer} {@code (value) -> controller.setP(value)}
+   *
+   * @param controller The PID controller that the mapping will target
+   * @param pidParam The {@link frc.utils.pid.PIDParam} that the mapping will reference
+   */
+  public static Consumer<Double> getCorrespondingSetMethod(
+      SparkPIDController controller, PIDParam pidParam) {
+
+    Consumer<Double> returnValue = null;
+
+    switch (pidParam) {
+      case P:
+        returnValue = x -> controller.setP(x);
+        break;
+      case I:
+        returnValue = x -> controller.setI(x);
+        break;
+      case D:
+        returnValue = x -> controller.setD(x);
+        break;
+      case I_ZONE:
+        returnValue = x -> controller.setIZone(x);
+        break;
+      case FF:
+        returnValue = x -> controller.setFF(x);
+        break;
+      case MAX_OUTPUT:
+        returnValue = x -> controller.setOutputRange(controller.getOutputMin(), x);
+        break;
+      case MIN_OUTPUT:
+        returnValue = x -> controller.setOutputRange(x, controller.getOutputMax());
+        break;
+    }
+
+    return returnValue;
+  }
+
+  /**
+   * Maps a {@link frc.utils.pid.PIDParam} to the corresponding getter method of a {@link
+   * com.revrobotics.SparkPIDController}.
+   *
+   * <p>Example: {@code getCorrespondingGetMethod(controller, PIDParam.P)} will return a {@link
+   * java.util.function.Supplier} {@code () -> controller.getP()}
+   *
+   * @param controller The PID controller that the mapping will target
+   * @param pidParam The {@link frc.utils.pid.PIDParam} that the mapping will reference
+   */
+  public static Supplier<Double> getCorrespondingGetMethod(
+      SparkPIDController controller, PIDParam pidParam) {
```

Comment:
Also check if current pid value being set is the same as the nt entry from shuffleboard, and if not return the shuffleboard value and call set. Not 100% sure how this will integrate with your current code, but I'll sleep on it. It's important for the boilerplate synch code with the user input also happens here to avoid subsystem ugliness.

Reply:
I'll switch this to an internal method and make a synced wrapper, as that would require an extra parameter for the Shuffleboard key. Shuffleboard sync might be out of scope for a util class though, given single-responsibility.